src/battle_controllers.c

```diff
@@ -0,0 +1,1551 @@
+    stringInfo->currentMove = gCurrentMove;
+    stringInfo->lastMove = gLastUsedMove;
+    stringInfo->lastItem = gLastUsedItem;
+    stringInfo->lastAbility = gLastUsedAbility;
+    stringInfo->scrActive = gBattleScripting.bank;
+    stringInfo->unk1605E = gBattleStruct->field_52;
+    stringInfo->hpScale = gBattleStruct->hpScale;
+    stringInfo->StringBank = gStringBank;
+    stringInfo->moveType = gBattleMoves[gCurrentMove].type;
+
+    for (i = 0; i < BATTLE_BANKS_COUNT; i++)
+        stringInfo->abilities[i] = gBattleMons[i].ability;
+    for (i = 0; i < TEXT_BUFF_ARRAY_COUNT; i++)
+    {
+        stringInfo->textBuffs[0][i] = gBattleTextBuff1[i];
+        stringInfo->textBuffs[1][i] = gBattleTextBuff2[i];
+        stringInfo->textBuffs[2][i] = gBattleTextBuff3[i];
+    }
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, sizeof(struct StringInfoBattle) + 4);
+}
+
+void EmitPrintStringPlayerOnly(u8 bufferId, u16 stringID)
+{
+    s32 i;
+    struct StringInfoBattle* stringInfo;
+
+    gBattleBuffersTransferData[0] = CONTROLLER_PRINTSTRINGPLAYERONLY;
+    gBattleBuffersTransferData[1] = 17;
+    gBattleBuffersTransferData[2] = stringID;
+    gBattleBuffersTransferData[3] = (stringID & 0xFF00) >> 8;
+
+    stringInfo = (struct StringInfoBattle*)(&gBattleBuffersTransferData[4]);
+    stringInfo->currentMove = gCurrentMove;
+    stringInfo->lastMove = gLastUsedMove;
+    stringInfo->lastItem = gLastUsedItem;
+    stringInfo->lastAbility = gLastUsedAbility;
+    stringInfo->scrActive = gBattleScripting.bank;
+    stringInfo->unk1605E = gBattleStruct->field_52;
+
+    for (i = 0; i < BATTLE_BANKS_COUNT; i++)
+        stringInfo->abilities[i] = gBattleMons[i].ability;
+    for (i = 0; i < TEXT_BUFF_ARRAY_COUNT; i++)
+    {
+        stringInfo->textBuffs[0][i] = gBattleTextBuff1[i];
+        stringInfo->textBuffs[1][i] = gBattleTextBuff2[i];
+        stringInfo->textBuffs[2][i] = gBattleTextBuff3[i];
+    }
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, sizeof(struct StringInfoBattle) + 4);
+}
+
+void EmitChooseAction(u8 bufferId, u8 arg1, u16 arg2)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_CHOOSEACTION;
+    gBattleBuffersTransferData[1] = arg1;
+    gBattleBuffersTransferData[2] = arg2;
+    gBattleBuffersTransferData[3] = (arg2 & 0xFF00) >> 8;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitCmd19(u8 bufferId)
+{
+    gBattleBuffersTransferData[0] = CONTROLLER_19;
+    gBattleBuffersTransferData[1] = 19;
+    gBattleBuffersTransferData[2] = 19;
+    gBattleBuffersTransferData[3] = 19;
+    PrepareBufferDataTransfer(bufferId, gBattleBuffersTransferData, 4);
+}
+
+void EmitChooseMove(u8 bufferId, bool8 isDoubleBattle, bool8 NoPpNumber, struct ChooseMoveStruct *movePpData)
+{
+    s32 i;
+
+    gBattleBuffersTransferData[0] = CONTROLLER_CHOOSEMOVE;
+    gBattleBuffersTransferData[1] = isDoubleBattle;
+    gBattleBuffersTransferData[2] = NoPpNumber;
```

Comment:
What does "NoPpNumber" mean exactly?

Reply:
It means it won't display a pp number when you select a move. In the games it's always set to FALSE.